templates/documents/single_collection_home.html

```diff
@@ -44,9 +44,19 @@ <h1 class="pg-title">{{ collection.name }} {% if collection.is_index %}<span cla
     <div class="flex flex-col gap-4">
       <div class="flex justify-between items-center">
         <h2 class="text-lg font-semibold">Files</h2>
-        <button class="btn btn-sm btn-primary" onclick="chooseFilesModal.showModal()">
-          <i class="fa-solid fa-plus"></i> Add Files
-        </button>
+        <div class="join">
+          {% if collection.has_failed_index_uploads %}
+            <form class="join-item" action="{% url 'documents:retry_failed_uploads' team.slug collection.id %}" method="POST">
```

Comment:
the user could click this multiple times and kick of multiple tasks - might be worth storing the task ID somewhere and checking it

Reply:
I think I'm going to change the way I did it slightly by passing in the collection files that I want to upload instead. This way, the caller will decide which files should be uploaded, preventing possible clashes if an upload and retry task is running at the same time

Reply:
See https://github.com/dimagi/open-chat-studio/pull/1606/commits/c5295170c0d5f88da2f5527772f19055b0e250dc